### ISSUE TITLE
Time estimate in the progressbar

### ISSUE THREAD

**Issue by ChrisRackauckas** [JunoLab/Juno.jl#3]
@pfitzseb opening this issue up here as this seems like the right place (is it?). What's the current state of this?

**Comment by pfitzseb** [comment]
The julia side of this is easy (and done), but I'm still not totally sure about the UI (my first idea didn't work out) -- haven't had any time for this the last week though.

**Comment by pfitzseb** [comment]
![progress](https://cloud.githubusercontent.com/assets/6735977/18417642/0b06585a-7836-11e6-89e4-fbbbbf6f36f8.png)
Okay, so this is what I have so far. 
One thing I thought would be nice is that a click on the progress bar will get you to the file where `@progress` was called. I tried getting the line as well, but that doesn't really work because of macro expansion and stuff (at least I think so).

Thoughts? Anything else you'd want?

**Comment by ChrisRackauckas** [comment]
I don't think getting to the file is too necessary. Maybe for someone's script sure, but it'll be buried in a package for my use, so most people probably won't want to do that. 

BTW, I'm hitting the `Atom.progress` function directly since I'm doing this in while loops and it lets me control the percentage directly. This is part of the function and not just the macro, right? Or if it's part of the macro, are there some functions I use to get to that display?

**Comment by pfitzseb** [comment]
Well, the implementation of the `@progress` macro is [here](https://github.com/pfitzseb/Juno.jl/commit/34cd5ff83df9b10eb035873d949dc6dc2e57057f). I guess I could make the `progress`-API nicer by using globals and stuff (kinda like `tic()`-`toc()`), but not sure if that's necessary or if it's okay to do the timing yourself?

**Comment by ChrisRackauckas** [comment]
As long as the Atom.progress function has all the tools to do it, it should be fine. So instead of giving the percentage `i/`n, now it's `i`, `n`, and the elapsed time? Then it uses the elapsed time and the percentage to show the estimated time left? That will work quite well. What's this `file`?

**Comment by pfitzseb** [comment]
`file` is the file which is shown in the tooltip ;)
The old method is still there as well, so I'll add another one for just `i`, `n` and `elapsed_time` in case you don't want to provide that.

**Comment by pfitzseb** [comment]
Sounds good?
```
    progress(i, n, t_elapsed, [file])

Set Atom's progress bar to `i/n` and calculate the remaining time from `t_elapsed`.
If the `file` argument is provided, the progress bar will be linked to said file.
```

**Comment by ChrisRackauckas** [comment]
Yeah, that looks great!

What do you think the timeline will be for merging this into master?

**Comment by pfitzseb** [comment]
Well, those changes should be mostly uncontroversial... so as soon as I have commit access to Juno.jl I can push this to master ;)

**Comment by ChrisRackauckas** [comment]
Solved by #19
